### PR TITLE
feat: add replicate.enable property to collection and related config

### PR DESCRIPTION
**Pull request by SimFG**
- issue: #37105

### CODE REVIEW

**Comment by SimFG** [comment]
rerun go-sdk

**Comment by SimFG** [comment]
rerun cpp-unit-test

**Comment by SimFG** [comment]
rerun go-sdk
/run-cpu-e2e
rerun cpp-unit-test

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

File: internal/proxy/task_insert.go [hunk 1]
```diff
@@ -125,6 +125,15 @@ func (it *insertTask) PreExecute(ctx context.Context) error {
 		return merr.WrapErrAsInputError(merr.WrapErrParameterTooLarge("insert request size exceeds maxInsertSize"))
 	}
 
+	info, err := globalMetaCache.GetCollectionInfo(it.ctx, it.insertMsg.GetDbName(), collectionName, 0)
```
Comment (xiaofan-luan) [inline internal/proxy/task_insert.go:128]: GetCollectionInfo && GetCollectionSchema twice might not be slow. and why do we need a Baisc Collection info?

Comment (xiaofan-luan) [reply internal/proxy/task_insert.go:128]: we can do getFullCollectionInfo so base info and schema is there

**Pull request closed without merging**